### DISCUSSION TITLE
Deadlock if SPI is shared with an SD-card

### DISCUSSION THREAD

**Discussion by eetnev** [zephyrproject-rtos/zephyr#71433] [General]
Hi,

I've configured both flash memory (mx25r64) and an SD-card on the same SPI bus and I'm encountering deadlock issues when the SD-card slot is left empty and the file system in the flash memory is being accessed.

It seems that [`sdhc_spi_request()`](https://github.com/zephyrproject-rtos/zephyr/blob/1ec6a94679aeabe2f6e8d96b5c585ff18c67a3ec/drivers/sdhc/sdhc_spi.c#L640) doesn't call `spi_release()` if `sdhc_spi_send_cmd()` or `sdhc_spi_write_data()` fails. Is there a specific reason for this? I need to call `disk_access_init()` to determine if the SD-card is present or not. Naturally, if the SD-card slot is empty, the function will fail with an `-ETIMEDOUT` error but now the SPI is left in locked state for the reasons stated above.

For the record, everything seems to run smoothly if an SD-card has been inserted.

**Comment by ndrs-pst** [comment]
@eetnev Just letting you know that this issue has already been addressed in https://github.com/zephyrproject-rtos/zephyr/pull/72754.
